src/audio/generic_processor/hifi_codec/codec.c

```diff
@@ -65,3 +65,28 @@ int hifi_codec_load_config(struct comp_dev *dev, void *cfg, size_t size,
 	dst->data = NULL;
 	return ret;
 }
+
+int hifi_codec_init(struct comp_dev *dev, uint32_t codec_id)
+{
+	int ret;
+
+	comp_info(dev, "hifi_codec_init() start");
+
+	/* select API */
+	if (codec_id >
```

Comment:
Is a single uuid shared by all codecs and there is another codec_id? So all instances share same uuid for logging etc?

Reply:
Yes UUID is single - only codec_adapter gets its own UUID. If you want to create a pipeline with two codecs then you add two codec_adapter components each working with different codec.